packages/core/src/utils/make-namespaced.ts

```diff
@@ -44,7 +44,16 @@ export function makeNamespacedInput(input: SolcInput, output: SolcOutput): SolcI
       const inherits = contractDef.baseContracts;
       for (const inherit of inherits) {
         if (isNodeType('InheritanceSpecifier', inherit)) {
-          modifications.push(getReplace(inherit, inherit.baseName.name ?? '')); // TODO handle when name is undefined (for UserDefinedTypeName)
+          const baseName = inherit.baseName;
+
+          let name;
+          if (isNodeType('UserDefinedTypeName', baseName)) {
+            name = (baseName.name ?? baseName.pathNode?.name)!;
+          } else {
+            name = baseName.name;
+          }
+
+          modifications.push(getReplace(inherit, name));
```

Comment:
This can be a lot simpler:
```suggestion
          assert(inherit.baseName.name !== undefined);
          modifications.push(getReplace(inherit, inherit.baseName.name));
```

See https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/876/files/db69587e61da520fb1949c86950bd029f0cdc17e#r1332135132 if you haven't seen it yet.